### PR TITLE
MOSIP-20419 Added /check-status/aid api.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: resident/resident-service/src/test/resources/application.properties [hunk 1]
```diff
@@ -336,6 +336,7 @@ mosip.scope.resident.getAuthLockStatus=${mosip.resident.identity.claim.individua
 mosip.scope.resident.patchUpdateUin=${mosip.resident.identity.claim.individual-id}
 mosip.scope.resident.postUploadDocuments=${mosip.resident.identity.claim.individual-id}
 mosip.scope.resident.getUploadedDocuments=${mosip.resident.identity.claim.individual-id}
+mosip.scope.resident.getCredentialRequestStatus=${mosip.resident.identity.claim.individual-id}
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/test/resources/application.properties]: Correct the property like `getCheckAidStatus`

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 2]
```diff
@@ -762,4 +762,12 @@ private void validateServiceType(String serviceType, String request_service_hist
 			}
 		}
 	}
+
+	public void validateCredentialRequestStatusRequest(String aid) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Correct the property like `validateCheckAidStatusRequest`

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java [hunk 3]
```diff
@@ -1355,4 +1363,27 @@ public AidStatusResponseDTO getAidStatus(AidStatusRequestDTO reqDto) throws Resi
 		}
 	}
 
+	@Override
+	public String getCredentialRequestStatus(String aid) throws ResidentServiceCheckedException {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: Correct the property like `getCheckAidStatus`

File: resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java [hunk 4]
```diff
@@ -471,6 +471,12 @@ public enum EventEnum {
 	IDA_TOKEN_NOT_FOUND("RES-SER-217", RegistrationConstants.SYSTEM, "IDA token not found",
 			"IDA token not found", "RES-SER", "Residence service", "NO_ID", "NO_ID_TYPE",
 			RegistrationConstants.APPLICATIONID, RegistrationConstants.APPLICATIONNAME),
+	REQ_CREDENTIAL_REQUEST_STATUS("RES-SER-218", RegistrationConstants.SYSTEM, "Request credential request status",
+			"Requesting credential request status is success", "RES-SER", "Residence service", "NO_ID", "NO_ID_TYPE",
+			RegistrationConstants.APPLICATIONID, RegistrationConstants.APPLICATIONNAME),
+	REQ_CREDENTIAL_REQUEST_STATUS_SUCCESS("RES-SER-219", RegistrationConstants.SYSTEM, "Request credential request status",
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java]: Correct the property like `CHECK_AID_STATUS_REQUEST_`

File: resident/resident-service/src/main/java/io/mosip/resident/service/ResidentService.java [hunk 5]
```diff
@@ -40,4 +40,9 @@ AidStatusResponseDTO getAidStatus(AidStatusRequestDTO reqDto)
     List<ServiceHistoryResponseDto> getServiceHistory(Integer pageStart, Integer pageFetch, LocalDateTime fromDateTime, LocalDateTime toDateTime, String serviceType, String sortType) throws ResidentServiceCheckedException, ApisResourceAccessException;
 
 	List<ResidentServiceHistoryResponseDto> getServiceRequestUpdate(Integer pageStart, Integer pageFetch) throws ResidentServiceCheckedException;
+
+	AidStatusResponseDTO getAidStatus(AidStatusRequestDTO reqDto, boolean performOtpValidation) throws ResidentServiceCheckedException, ApisResourceAccessException, OtpValidationFailedException ;
+
+	String getCredentialRequestStatus(String aid) throws ResidentServiceCheckedException;
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/ResidentService.java]: Correct the property like `checkAidStatus`

File: resident/resident-service/src/main/java/io/mosip/resident/dto/AuthorizedScopesDto.java [hunk 6]
```diff
@@ -40,5 +40,7 @@ public class AuthorizedScopesDto {
 	private List<String> getServiceAuthHistoryRoles;
 
 	private List<String> getServiceRequestUpdate;
+
+	private List<String> getCredentialRequestStatus;
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/dto/AuthorizedScopesDto.java]: Correct the property like `getCheckAidStatus`

File: resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentController.java [hunk 7]
```diff
@@ -229,6 +229,34 @@ public ResponseWrapper<AuthHistoryResponseDTO> reqAuthHistory(
...
+			+ ")")
+	@GetMapping(path="/check-status/aid/{AID}")
+	@Operation(summary = "getCredentialRequestStatus", description = "getCredentialRequestStatus", tags = { "resident-controller" })
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "OK"),
+			@ApiResponse(responseCode = "201", description = "Created", content = @Content(schema = @Schema(hidden = true))),
+			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content(schema = @Schema(hidden = true))),
+			@ApiResponse(responseCode = "403", description = "Forbidden", content = @Content(schema = @Schema(hidden = true))),
+			@ApiResponse(responseCode = "404", description = "Not Found", content = @Content(schema = @Schema(hidden = true))) })
+	public ResponseWrapper<ResponseDTO> getCredentialRequestStatus(
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentController.java]: Correct the property like `checkAidStatus`

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/test/resources/application.properties (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/test/resources/application.properties]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/EventEnum.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/service/ResidentService.java (hunk 5 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/ResidentService.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/dto/AuthorizedScopesDto.java (hunk 6 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/dto/AuthorizedScopesDto.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentController.java (hunk 7 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/controller/ResidentController.java]: Done.

**Review by loganathan-sekaran — APPROVED** [review]

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**